### PR TITLE
perf (MonoidAlgebra.Basic): reduce withs in structure terms

### PR DESCRIPTION
**Pull request by mattrobball**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by mattrobball** [comment]
!bench

File: Mathlib/Algebra/MonoidAlgebra/Basic.lean [hunk 1]
```diff
@@ -281,30 +278,33 @@ section Semiring
...
     MonoidAlgebra k G →+* R :=
-  { liftNC (f : k →+ R) g with
-    toFun := liftNC (f : k →+ R) g
-    map_one' := liftNC_one _ _
-    map_mul' := fun _a _b => liftNC_mul _ _ _ _ fun {_ _} _ => h_comm _ _ }
+  -- { toMonoidHom :=liftNC (f : k →+ R) g with
+  { toMonoidHom := {
+      toFun := liftNC (f : k →+ R) g
+      map_one' := liftNC_one _ _
+      map_mul' := fun _a _b => liftNC_mul _ _ _ _ fun {_ _} _ => h_comm _ _ }
```
Comment (eric-wieser) [inline Mathlib/Algebra/MonoidAlgebra/Basic.lean:298]: Doesn't this generate an identical term?

Comment (mattrobball) [reply Mathlib/Algebra/MonoidAlgebra/Basic.lean:298]: I don't think so but didn't check

**Comment by mattrobball** [comment]
!bench

**Comment by semorrison** [comment]
(Just a reminder that on the mathlib4 repo, if authors don't self-label with `awaiting-review`, PRs don't appear on the [#queue](https://github.com/leanprover-community/mathlib4/pulls?q=is:open+label:awaiting-review+sort:updated-asc+-label:blocked-by-other-PR+-label:merge-conflict+-status:failure+-label:awaiting-CI) and might never be noticed. :-)

**Pull request closed without merging**